pacman/game.py

```diff
@@ -212,10 +217,22 @@ def draw_state(self, title):
             self.plot = plt.matshow(current_board, fignum=0)
         plt.pause(self.speed)
 
-    def get_state(self):
+    def get_state(self, representation='board'):
         """Returns a hashable representation of the current state"""
         # TODO: Ghosts hide candies in this state representation
-        hashable_state = tuple(self.compute_grid().flatten())
+        if(representation == 'board'):
+            hashable_state = tuple(self.compute_grid().flatten())
+        elif(representation == 'features'):
+            pacman_pos = np.asarray(self.pacman.position)
+            # Get closest ghost relativ position
+            ghost_positions = [np.asarray(ghost.position)
+                               for ghost in self.ghosts]
+            ghost_distances = [np.linalg.norm(ghost_pos - pacman_pos)
+                               for ghost_pos in ghost_positions]
+            idx_closest_ghost = np.argmin(ghost_distances)
+            ghost_pos = ghost_positions[idx_closest_ghost]
+            ghost_rel_pos = ghost_pos - pacman_pos
+            hashable_state = (tuple(ghost_rel_pos),)
```

Comment:
Top !
We could refactor this later to a method to be used for the other features (closest pac dots, closest candy).
The method could take as argument a list of objects positions (either all the ghosts, all the pac-dots ...) and return the relative position of the closest one).